Profile payload refactor

phmccarty opened the pull request:
Move the duplicated code into a new function, and have all of the patterns use it if they support it. Also, add a stub for the `make` pattern.

bryteise commented on autospec/specfiles.py:
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
+            return
+        if not config.profile_payload:
+            return
+        if config.profile_payload and not config.profile_payload[0]:
```
extra check on config.profile_payload

phmccarty replied:
fixed

bryteise commented on autospec/specfiles.py:
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
+        else:
+            print_warning(f"profile_payload not supported for {pattern} pattern")
+            return
+        if config.profile_payload and config.profile_payload[0]:
```
extra check on config.profile_payload

phmccarty replied:
fixed

bryteise commented on autospec/specfiles.py:
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
+        if config.profile_payload and config.profile_payload[0]:
+            if use_subdir and self.subdir:
+                self._write_strip("pushd " + self.subdir)
+            self._write_strip(init)
```
since you check for post, maybe check for init too.

phmccarty replied:
fixed in v2

bryteise commented on autospec/specfiles.py:
```diff
@@ -572,6 +573,55 @@ def write_license_files(self):
+            init = "{0}".format(self.get_profile_generate_flags())
+            post = "{0}".format(self.get_profile_use_flags())
+        else:
+            print_warning(f"profile_payload not supported for {pattern} pattern")
```
maybe this should be a hard error for testing

phmccarty replied:
I removed this block in v2

bryteise commented:
Seems like a reasonable approach.

phmccarty commented:
Changes since the v1 PR:
- Do not special-case the `make` pattern. Instead, if `pattern` is not passed to `write_profile_payload()` (or is `None`), the default behavior is assumed and no special init/post steps are needed.
- Switch to concatenated formatted string literals instead of calling `format`

bryteise commented on autospec/specfiles.py:
```diff
@@ -572,6 +572,50 @@ def write_license_files(self):
+        if not config.profile_payload:
+            return
+        if not config.profile_payload[0]:
+            return
```
if not config.profile_payload or not config.profile_payload[0] probably but up to you.

phmccarty replied:
I will remove the second conditional, since we determined that it's not needed.
